checker/jdk/nullness/src/java/util/logging/LogRecord.java

```diff
@@ -0,0 +1,592 @@
+     * may not map to any operating system ID.
+     *
+     * @return thread ID
+     */
+    public int getThreadID() {
+        return threadID;
+    }
+
+    /**
+     * Set an identifier for the thread where the message originated.
+     * @param threadID  the thread ID
+     */
+    public void setThreadID(int threadID) {
+        this.threadID = threadID;
+    }
+
+    /**
+     * Get event time in milliseconds since 1970.
+     *
+     * @return event time in millis since 1970
+     */
+    public long getMillis() {
+        return millis;
+    }
+
+    /**
+     * Set event time.
+     *
+     * @param millis event time in millis since 1970
+     */
+    public void setMillis(long millis) {
+        this.millis = millis;
+    }
+
+    /**
+     * Get any throwable associated with the log record.
+     * <p>
+     * If the event involved an exception, this will be the
+     * exception object. Otherwise null.
+     *
+     * @return a throwable
+     */
+    public @Nullable Throwable getThrown() {
+        return thrown;
+    }
+
+    /**
+     * Set a throwable associated with the log event.
+     *
+     * @param thrown  a throwable (may be null)
+     */
+    public void setThrown(@Nullable Throwable thrown) {
+        this.thrown = thrown;
+    }
+
+    private static final long serialVersionUID = 5372048053134512534L;
+
+    /**
+     * @serialData Default fields, followed by a two byte version number
+     * (major byte, followed by minor byte), followed by information on
+     * the log record parameter array.  If there is no parameter array,
+     * then -1 is written.  If there is a parameter array (possible of zero
+     * length) then the array length is written as an integer, followed
+     * by String values for each parameter.  If a parameter is null, then
+     * a null String is written.  Otherwise the output of Object.toString()
+     * is written.
+     */
+    @SuppressWarnings("dereference.of.nullable") // parameters and parameters[i] is always ensured to be non-null at every dereference
+    private void writeObject(ObjectOutputStream out) throws IOException {
+        // We have to call defaultWriteObject first.
+        out.defaultWriteObject();
+
+        // Write our version number.
+        out.writeByte(1);
+        out.writeByte(0);
+        if (parameters == null) {
+            out.writeInt(-1);
+            return;
+        }
+        out.writeInt(parameters.length);
+        // Write string values for the parameters.
+        for (int i = 0; i < parameters.length; i++) {
+            if (parameters[i] == null) {
+                out.writeObject(null);
+            } else {
+                out.writeObject(parameters[i].toString());
+            }
+        }
+    }
+
+    @SuppressWarnings({"dereference.of.nullable" , "argument.type.incompatible"}) // method first initializes parameters before its dereference and ClassLoader.getSystemClassLoader() can't be @Nullable here
```

Comment:
1. The comment is misleading.  The Nullness Checker handles the initialization correctly and no warning suppression is needed.  The issue is that the Nullness Checker can't determine that `parameters` *stays* non-null.
Therefore, I would write the comment more like
``` // in.readObject does not affect parameters field.  http://tinyurl.com/cfissue/984```

2. How do you know that `ClassLoader.getSystemClassLoader()` can't be null?  Please explain that.

3. Minor:  "can't be `@Nullable`" is incorrect terminology, since the expression does have `@Nullable` type.  You want to claim that it also has `@NonNull` type, which is a more specific fact.

Reply:
> 2. How do you know that `ClassLoader.getSystemClassLoader()` can't be null?  Please explain that.

My reasoning is that all the logging framework will definitely have a ClassLoader which will also act as SystemClassLoader. I now have mentioned this in comments too.
I could be wrong.
Please let me know if my reasoning is wrong because without `ClassLoader.getSystemLoader()` being `@NonNull` it would straight up produce `NullPointerException`.